dlc/src/main/scala/org/bitcoins/dlc/execution/DLCExecutor.scala

```diff
@@ -116,6 +143,20 @@ case class DLCExecutor(signer: DLCTxSigner)(implicit ec: ExecutionContext) {
 
 object DLCExecutor {
 
+  def findOutcomeAndSignatures(
+      contractInfo: ContractInfo,
+      oracleSigs: Vector[OracleSignatures]): Option[
+    (OracleOutcome, Vector[OracleSignatures])] = {
+    val threshold = contractInfo.oracleInfo.threshold
+    val sigCombinations = CETCalculator.combinations(oracleSigs, threshold)
+    sigCombinations.flatMap { sigs =>
+      contractInfo.findOutcome(sigs) match {
+        case Some(outcome) => Some((outcome, sigs))
+        case None          => None
+      }
+    }.headOption // should only be one
```

Comment:
in that case this should either be re-written to use an `Option` `var` updated inside of a `find/takeWhile/dropWhile` or else a tail recursive function that goes through `sigCombinations` but returns if it finds something because currently you go through all of `sigCombinations` with `findOutcome` the first of which scales exponentially and the second of which is also a really really costly function call.